python/lancedb/embeddings/base.py

```diff
@@ -125,6 +137,12 @@ def VectorField(self, **kwargs):
         """
         return Field(json_schema_extra={"vector_column_for": self}, **kwargs)
 
+    def __eq__(self, __value: object) -> bool:
```

Comment:
where are you using eq/hash btw?

Reply:
oic you're using it with the weak references.

Reply:
yea